### PR TITLE
Closes #7710: Stop intercepting redirect with subframe load request without user action

### LINKED ISSUE #7710: Should not intercept redirect with subframe load request without user action
Currently `onSubframeLoadRequest` can trigger redirect to external app without user action.  Redirect to external app was no meant for subframes.  Add checks so if the redirect is coming from a subframe without user action, do not redirect to external apps.

### PR DESCRIPTION
**Pull request by rocketsroger**
---

### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/master/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Milestone**: Make sure issues closed by this pull request are added to the [milestone](https://github.com/mozilla-mobile/android-components/milestones) of the version currently in development.
- **Breaking Changes**: If this is a breaking change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### CODE REVIEW

**Comment by rocketsroger** [comment]
Fenix side change https://github.com/mozilla-mobile/fenix/pull/12566

File: components/concept/engine/src/main/java/mozilla/components/concept/engine/request/RequestInterceptor.kt [hunk 1]
```diff
@@ -74,7 +74,8 @@ interface RequestInterceptor {
         hasUserGesture: Boolean,
         isSameDomain: Boolean,
         isRedirect: Boolean,
-        isDirectNavigation: Boolean
+        isDirectNavigation: Boolean,
+        isSubframeRequest: Boolean
```
Comment (csadilek) [inline components/concept/engine/src/main/java/mozilla/components/concept/engine/request/RequestInterceptor.kt:78]: OK, only request is here is to update the KDocs :) to add the new `isSubframeRequest` but also make it clearer what the difference is between `hasUserGesture`, `isDirectNavigation` and `isSubframeRequest`. I don't think I even know :).

Comment (rocketsroger) [reply components/concept/engine/src/main/java/mozilla/components/concept/engine/request/RequestInterceptor.kt:78]: My mistake.  I'll add the parameter to the KDocs. 👍

**Comment by rocketsroger** [comment]
bors r=csadilek

**Comment by rocketsroger** [comment]
Reference browser side fix https://github.com/mozilla-mobile/reference-browser/pull/1267

**Pull request merged**